### ISSUE TITLE
[BUG] java.lang.IllegalArgumentException: Invalid version for API key API_VERSIONS: 3

### ISSUE THREAD

**Issue by sido420** [streamnative/kop#166]
**Describe the bug**
Pulsar starts fine with Kafka protocol handler installed however I see error messages in logs.

```
13:12:48.091 [ForkJoinPool.commonPool-worker-3] INFO  org.apache.pulsar.broker.service.ServerCnx - [/127.0.0.1:56110] Created subscription on topic persistent://public/default/exclamation-input / public/default/exclamation
13:12:57.442 [pulsar-io-25-13] INFO  io.streamnative.pulsar.handlers.kop.KafkaRequestHandler - channel inactive [id: 0x17ea1bee, L:/127.0.0.1:9092 ! R:/127.0.0.1:56113]
13:12:57.443 [pulsar-io-25-13] INFO  io.streamnative.pulsar.handlers.kop.KafkaRequestHandler - close channel [id: 0x17ea1bee, L:/127.0.0.1:9092 ! R:/127.0.0.1:56113]
13:12:58.158 [pulsar-io-25-3] INFO  io.streamnative.pulsar.handlers.kop.KafkaRequestHandler - channel active: [id: 0x62a5187c, L:/127.0.0.1:9092 - R:/127.0.0.1:56128]
13:12:58.159 [pulsar-io-25-3] ERROR io.streamnative.pulsar.handlers.kop.KafkaCommandDecoder - error to get Response ByteBuf:
java.lang.IllegalArgumentException: Invalid version for API key API_VERSIONS: 3
	at org.apache.kafka.common.protocol.ApiKeys.schemaFor(ApiKeys.java:312) ~[?:?]
	at org.apache.kafka.common.protocol.ApiKeys.responseSchema(ApiKeys.java:286) ~[?:?]
	at org.apache.kafka.common.requests.ApiVersionsResponse.toStruct(ApiVersionsResponse.java:129) ~[?:?]
	at org.apache.kafka.common.requests.ResponseUtils.serializeResponse(ResponseUtils.java:40) ~[?:?]
	at io.streamnative.pulsar.handlers.kop.KafkaCommandDecoder.responseToByteBuf(KafkaCommandDecoder.java:112) ~[?:?]
	at io.streamnative.pulsar.handlers.kop.KafkaCommandDecoder.writeAndFlushResponseToClient(KafkaCommandDecoder.java:238) ~[?:?]
	at io.streamnative.pulsar.handlers.kop.KafkaCommandDecoder.lambda$channelRead$0(KafkaCommandDecoder.java:209) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760) [?:1.8.0_121]
	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:778) [?:1.8.0_121]
	at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2140) [?:1.8.0_121]
	at io.streamnative.pulsar.handlers.kop.KafkaCommandDecoder.channelRead(KafkaCommandDecoder.java:208) [pulsar-protocol-handler-kafka-0.3.0-20200628-rc.nar-unpacked/:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:321) [io.netty-netty-codec-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:295) [io.netty-netty-codec-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) [io.netty-netty-transport-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [io.netty-netty-common-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty-netty-common-4.1.48.Final.jar:4.1.48.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty-netty-common-4.1.48.Final.jar:4.1.48.Final]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
```

**To Reproduce**
Here is what I did using lates Pulsar and KoP:
- Installed KoP using instructions in documentation.
- Tried the Kafka consumer and produce test.
- Here are relevant settings in broker.conf:

```
# Hostname or IP address the service binds on, default is 0.0.0.0.
bindAddress=0.0.0.0

# Hostname or IP address the service advertises to the outside world. If not set, the value of InetAddress.getLocalHost().getHostName() is used.
advertisedAddress=127.0.0.1

messagingProtocols=kafka
protocolHandlerDirectory=./protocol_handlers
allowAutoTopicCreationType=partitioned
# Use the same host value set for 'advertisedAddress' above (comma separated list is possible but must reflect advertisedAddress)
#listeners=PLAINTEXT://127.0.0.1:9092,PLAINTEXT://192.168.1.31:9092
listeners=PLAINTEXT://127.0.0.1:9092,SSL://127.0.0.1:9093
```

**Expected behavior**
Expected it to just work

**Comment by sijie** [comment]
@sido420 what is the Kafka version you are using?

**Comment by sido420** [comment]
```
kafka_2.11-2.0.0
apache-pulsar-2.6.0
apache-zookeeper-3.6.1-bin
```

**Comment by sido420** [comment]
However, errors may be triggered by [Materialized](https://materialize.io/) that has a kafka source option. I dont see errors when i kill materialized.

If you can tell me the issue, I will open a ticket on their repo.

**Comment by sijie** [comment]
@sido420 I see. Do you know what version of Kafka library that Materialized is using? I am guessing it is using either a newer version or an older version of Kafka.

**Comment by sido420** [comment]
Looks like librdkafka is being used. Here is the version:

`librdkafka v1.4.2`

**Comment by sijie** [comment]
@sido420 I don't think we support librdkafka v1.4.2 yet. It is in our roadmap. If you are interested in contributing to supporting this version of librdkafka, contributions are welcome!

**Comment by sido420** [comment]
Are older/newer versions of librdkafka supported? If yes, which ones?

I might be help to help but will require some handholding to get started.

**Comment by sijie** [comment]
@sido420 Currently we only support v1.2.2. I think v1.4.2 is a newer version.

**Comment by sido420** [comment]
@sijie I got below from Materialize which makes sense to me and I think its going to be a quick change for KoP. What do you think?

>> I really think it'll be easier to attack this problem from the KoP angle. Note that librdkafka supports connecting to old Kafka brokers, all the way back to pre-1.0 Kafka brokers. KoP doesn't actually need to support the newest APIs; it just needs to not panic upon seeing a new version of an API that it doesn't support, but just return "unsupported" and masquerade as an older version of Kafka.

**Comment by sijie** [comment]
@sido420 Yeah, I was following that Github Issue. It makes sense to us. We will prioritize an improvement for it.

**Comment by sijie** [comment]
This has been fixed by #176.

@sido420 Can you try it again?
